### PR TITLE
[New Feature] Create new utilities : Currency Converter #286

### LINKED ISSUE #286: Create new utilities : Currency Converter
### What to do?
- Go to `utilities` from the navigation drawer
- Add a new section to it `Currency converter`
- On clicking that, open a new activity
- Show UI something like this
screen shot 2018-07-17 at 3 03 04 pm

- The country names can be a spinner. Keep the textbox below first country name to be editable. When value changes in that box, correspondingly change the value in text box below second country name.
- Don't add the last updated field, API doesn't provide that yet
- Show no internet animation, in case of no internet

### Getting list of currencies & currency code in java
https://stackoverflow.com/a/3537085/3623128

### Endpoint to use
`get-currency-conversion-rate/<str:source_currency_code>/<str:target_currency_code>`

### PR DESCRIPTION
**Pull request by santoshhiremani**
## Description

New Feature(Create new utilities : Currency Converter) fixes #286

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Android/app/src/main/assets/currencies.json [hunk 1]
```diff
@@ -0,0 +1 @@
...an Dollar","currencySymbol":"$","id":"NAD"},"NOK":{"currencyName":"Norwegian Krone","currencySymbol":"kr","id":"NOK"},"PLN":{"currencyName":"Polish Zloty","currencySymbol":"zł","id":"PLN"},"RUB":{"currencyName":"Russian Ruble","currencySymbol":"руб","id":"RUB"},"SZL":{"currencyName":"Swazi Lilangeni","id":"SZL"},"TJS":{"currencyName":"Tajikistani Somoni","id":"TJS"},"TTD":{"currencyName":"Trinidad and Tobago Dollar","currencySymbol":"TT$","id":"TTD"},"UGX":{"currencyName":"Ugandan Shilling","currencySymbol":"USh","id":"UGX"},"UYU":{"currencyName":"Uruguayan Peso","currencySymbol":"$U","id":"UYU"},"VND":{"currencyName":"Vietnamese Dong","currencySymbol":"₫","id":"VND"},"TND":{"currencyName":"Tunisian Dinar","id":"TND"},"UAH":{"currencyName":"Ukrainian Hryvnia","currencySymbol":"₴","id":"UAH"},"UZS":{"currencyName":"Uzbekistani Som","currencySymbol":"лв","id":"UZS"},"TMT":{"currencyName":"Turkmenistan Manat","id":"TMT"},"GBP":{"currencyName":"British Pound","currencySymbol":"£","id":"GBP"},"ZMW":{"currencyName":"Zambian Kwacha","id":"ZMW"},"BTC":{"currencyName":"Bitcoin","currencySymbol":"BTC","id":"BTC"},"BYN":{"currencyName":"New Belarusian Ruble","currencySymbol":"p.","id":"BYN"}}
```
Comment (Swati4star) [inline Android/app/src/main/assets/currencies.json]: Do we really need this json file? Check the issue description, I have already added a link to get currency code. You can get that by default in java.

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ActivityConversionListview.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+import butterknife.BindView;
+import io.github.project_travel_mate.R;
+import objects.CurrencyName;
+
+public class ActivityConversionListview extends Activity {
+
+    @BindView(R.id.listView)
+    ListView listview;
+    CurrencyConverterAdapter adaptr_listview;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/ActivityConversionListview.java]: We have used camelcase for variable names everywhere. You also use those only

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyActivity.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+        button_convert.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                from_amount = from_edittext.getText().toString();
+                convertCurrency();
+            }
+        });
+    }
+
+    private void convertCurrency() {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyActivity.java]: Add comments to the functions.

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyActivity.java [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        currences_names = new ArrayList<CurrencyName>();
+        mHandler = new Handler(Looper.getMainLooper());
+        mContext = this;
+
+        mDialog = new ProgressDialog(this);
+
+        first_country_short = "USD";
+        second_country_short = "INR";
+
+        from_layout.setOnClickListener(new View.OnClickListener() {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyActivity.java]: You can use butterknife for this.

File: Android/app/src/main/assets/currencies.json (hunk 1 above)
Comment (santoshhiremani) [reply Android/app/src/main/assets/currencies.json]: Yes, I checked that, it returns only list of currency codes. We need currency name as well.

**Comment by Swati4star** [comment]
@santoshhiremani Also, instead of pngs, can we have vector drawables?

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyActivity.java (hunk 4 above)
Comment (santoshhiremani) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyActivity.java]: These are just global variables. from_layout has it.

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyActivity.java (hunk 3 above)
Comment (santoshhiremani) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyActivity.java]: Done

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ActivityConversionListview.java (hunk 2 above)
Comment (santoshhiremani) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/ActivityConversionListview.java]: Done

**Comment by santoshhiremani** [comment]
Yes have to, As we have one more issue #414. Will fix in this.

**Comment by Swati4star** [comment]
@santoshhiremani Sure. We can have these fixed in that issue only.

**Pull request merged**